src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -158,6 +158,33 @@ public void MultipleServiceCanBeIEnumerableResolved()
         }
 
         [Fact]
+        public void RegistrationOrderIsPreservedWhenServicesAreIEnumerableResolved()
```

Comment:
There's nothing expectation we have with our DI system viz when requesting a single service, the most recently registered service wins. Could you add a test for that scenario as well?